presto-spark-base/src/main/java/com/facebook/presto/spark/execution/nativeprocess/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -149,6 +102,56 @@ public void stop()
         }
     }
 
+    void doGetTaskInfo()
```

Comment:
Nice refactor.

> Extract the code for making http call to fetch taskInfo
> into a separate to make it testable

There is a typo in " into a separate to make".

The commit message suggest that there will be a test that uses the newly extract method, but there is none. Is this intentional? 

It would be nice to comment that this method is intentionally "protected" to allow for testing. Consider adding VisibleForTesting annotation.

Reply:
> The commit message suggest that there will be a test that uses the newly extract method, but there is none. Is this intentional?

The function is use here -> https://github.com/prestodb/presto/pull/20265/files#diff-8549d10244ab0a9850d89bad95f687cb5f538241ad04da1913ff8081004a4f88R76

Reply:
Addressed these comments